test/Dialect/FIRRTL/annotations.fir

```diff
@@ -220,24 +220,24 @@ circuit Foo: %[[{"a":"{\"b\":null}"}]]
 ; // -----
 
 ; COM: Subfield/Subindex annotations should be parsed correctly on wires
-circuit Foo: %[[{"one":null,"target":"~Foo|Foo>bar.qux"},{"two":null,"target":"~Foo|Foo>bar[1].baz"} ]]
```

Comment:
Is `one` not legal ? Seems like we can't annotate on `qux` without indexing into bar.

Reply:
Yea, this is illegal according to @seldridge. The current `-lower-types` will ignore this annotation in the lowering. With this patch we'll emit an error on this.

Reply:
Right, that makes sense, Thanks